### PR TITLE
Add phoenix

### PR DESCRIPTION
**Pull request by waiteb3**
Because I promised

### CODE REVIEW

File: toolset/setup/linux/languages/elixir.sh [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
 
-# TODO seems to be broken installation
-echo "WARN: Elixir does not install"
-return 1
+RETCODE=$(fw_exists ${IROOT}/elixir.installed)
+[ ! "$RETCODE" == 0 ] || { return 0; }
 
-# fw_exists v0.13.3.tar.gz
-# [ $? -ne 0 ] || { return 0; }
+wget http://packages.erlang-solutions.com/erlang-solutions_1.0_all.deb
```
Comment (msmith-techempower) [inline toolset/setup/linux/languages/elixir.sh]: Should this be moved into `toolset/setup/linux/languages/erlang.sh` so other erlang frameworks can simply add it as a dependency if they aren't using elixir?

Comment (waiteb3) [inline toolset/setup/linux/languages/elixir.sh]: This is just `erlang.sh` with the elixir install added (line 11).

Comment (msmith-techempower) [inline toolset/setup/linux/languages/elixir.sh]: Then why is it here? Specify that `elixir` requires `erlang` via `fw_depends erlang` in an early line.

**Comment by waiteb3** [comment]
Cleaned it up.

File: toolset/setup/linux/languages/elixir.sh [hunk 2]
```diff
@@ -1,16 +1,9 @@
...
-# TODO seems to be broken installation
-echo "WARN: Elixir does not install"
-return 1
+RETCODE=$(fw_exists ${IROOT}/elixir.installed)
+[ ! "$RETCODE" == 0 ] || { return 0; }
 
-# fw_exists v0.13.3.tar.gz
-# [ $? -ne 0 ] || { return 0; }
+fw_depends erlang
+sudo apt-get install -y elixir
```
Comment (hamiltont) [inline toolset/setup/linux/languages/elixir.sh]: This should be a download, compile, install into `IROOT` instead - we're trying to completely remove sudo

File: frameworks/Elixir/phoenix/setup.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+# sed -i 's|db_host: "localhost",|db_host: "${DBHOST}",|g' config/config.exs
```
Comment (hamiltont) [inline frameworks/Elixir/phoenix/setup.sh]: Did you mean to leave this commented out?

**Comment by hamiltont** [comment]
Awesome, thanks! I've left some minor comments

File: toolset/setup/linux/languages/elixir.sh (hunk 2 above)
Comment (msmith-techempower) [inline toolset/setup/linux/languages/elixir.sh]: Agreed.

File: frameworks/Elixir/phoenix/setup.sh (hunk 3 above)
Comment (waiteb3) [inline frameworks/Elixir/phoenix/setup.sh]: Yes. It doesn't do any DB tests yet (the ORM is still pretty young).

Comment (hamiltont) [inline frameworks/Elixir/phoenix/setup.sh]: Gotcha, just checking

Comment (hamiltont) [inline frameworks/Elixir/phoenix/setup.sh]: Perhaps delete the line if it's unnecessary if you're making more changes to the PR

File: toolset/setup/linux/languages/elixir.sh (hunk 2 above)
Comment (waiteb3) [inline toolset/setup/linux/languages/elixir.sh]: If I promise to figure it out for the next round (erlang as well), can I be lazy today?

Comment (LadyMozzarella) [inline toolset/setup/linux/languages/elixir.sh]: This would be merged in for Round 11. Does that qualify as the next round?

**Comment by waiteb3** [comment]
Actually, I'll issue another PR when I get the ORM and other stuff figured out. Kane says it won't get into this round.

**Comment by msmith-techempower** [comment]
kain*

Of course it's not making it into this round [10].

**Comment by hamiltont** [comment]
FWIW, this [passed](https://travis-ci.org/TechEmpower/FrameworkBenchmarks/jobs/56779608) all verification checks, so if `elixir.sh` was not using sudo I'd be happy to merge this

**Comment by hamiltont** [comment]
So how about this: 

````
# Download and install without sudo
apt-get download elixir
dpkg -x elixir_*.deb $IROOT/elixir

# Binaries are now located here
EROOT=$IROOT/elixir/usr/local/bin/elixir

# Libraries are here
ELIBS=$IROOT/elixir/usr/local/lib/elixir
```

**Comment by msmith-techempower** [comment]
I like this, but they MUST have a permalink to a deb somewhere outside of using `apt-get`, right?

**Comment by hamiltont** [comment]
Yup, i'm sure that's the case. This is just an easy option to find the current official deb ;-)

**Pull request closed without merging**